File: cypress/support/data-loader.ts
/**
 * cypress/support/data-loader.ts — CP-SUPIN-05 fixture loader
 *
 * Per _specs/CROSS-FRAMEWORK-DATA-SHARING-v0.1-CS.md §4.1.
 *
 * Usage in Cypress specs:
 *   import { loadFixture, covers } from "../support/data-loader";
 *
 *   it("my test", () => {
 *     covers("TT-FUNC-001", "TT-LOV-insuranceCompanies");
 *     loadFixture<Participants>("test-participants").then(data => {
 *       const phoneA = data.participants.A.phone;
 *       // ...
 *     });
 *   });
 *
 * The cy.task('loadFixture', name) is registered in cypress.config.ts
 * setupNodeEvents and reads from fixtures/test-data/ at project root.
 *
 * DO NOT load fixtures from cypress/fixtures/ — that directory is intentionally
 * empty; all test data lives in the shared fixtures/test-data/ tree.
 */

// ─── loadFixture ────────────────────────────────────────────────────────────

/**
 * Load a YAML fixture by base name (without .yaml extension).
 * Returns a Cypress Chainable that resolves to the parsed object.
 *
 * @example
 *   loadFixture<{participants: Participants}>("test-participants").then(d => { ... })
 */
export function loadFixture<T = any>(name: string): Cypress.Chainable<T> {
  return cy.task<T>("loadFixture", name);
}

// ─── covers() annotation ─────────────────────────────────────────────────────

/**
 * Metadata annotation — records which TestTarget (TT) codes are exercised.
 * Logs to Cypress runner output; no effect on pass/fail.
 *
 * Per _specs/CROSS-FRAMEWORK-PARITY-EXECUTION-v0.1-CS.md §7.1 acceptance #2.
 *
 * @example
 *   covers("TT-LOV-vehicleBrands", "TT-SCRN-rozcestnik");
 */
export function covers(...ttCodes: string[]): void {
  Cypress.log({
    name: "covers",
    message: ttCodes.join(", "),
    consoleProps: () => ({ "TT codes": ttCodes }),
  });
}

// ─── Type exports (mirrors fixtures/test-data/*.yaml shapes) ─────────────────

export interface Participant {
  name: string;
  surname: string;
  op_number: string;
  rp_number?: string;
  birth_date: string;
  email: string;
  phone: string;
  phone_prefix: string;
  address_query: string;
  gdpr_consent: boolean;
  otp_code: string;
}

export interface Vehicle {
  spz: string;
  brand: string;
  model: string;
  insurer: string;
  color: string;
  desc: string;
}

export interface Participants {
  participants: Record<string, Participant>;
}

export interface Vehicles {
  vehicles: Record<string, Vehicle>;
}

File: selenium/helpers/data_loader.py
"""
selenium/helpers/data_loader.py — CP-SUPIN-05 fixture loader

Per _specs/CROSS-FRAMEWORK-DATA-SHARING-v0.1-CS.md §4.2.

Usage:
    from selenium.helpers.data_loader import load_fixture, covers

    data = load_fixture("test-participants")
    phone_a = data["participants"]["A"]["phone"]

    covers("TT-FUNC-001", "TT-LOV-insuranceCompanies")
"""
from __future__ import annotations

import os
from pathlib import Path
from typing import Any

import yaml

# Project root → fixtures/test-data/
FIXTURES_ROOT = Path(__file__).resolve().parents[3] / "fixtures" / "test-data"


def load_fixture(name: str) -> dict[str, Any]:
    """Load a YAML fixture by base name (without .yaml extension).

    Args:
        name: bare fixture name, e.g. "test-participants" or "test-vehicles"

    Returns:
        Parsed YAML object as a Python dict.

    Raises:
        FileNotFoundError: if the fixture file is not found at FIXTURES_ROOT.
    """
    base = name.removesuffix(".yaml")
    path = FIXTURES_ROOT / f"{base}.yaml"
    if not path.exists():
        raise FileNotFoundError(
            f"Fixture not found: {path}\n"
            f"FIXTURES_ROOT = {FIXTURES_ROOT}\n"
            f"Available: {sorted(p.name for p in FIXTURES_ROOT.glob('*.yaml'))}"
        )
    with open(path, encoding="utf-8") as fh:
        return yaml.safe_load(fh)


def covers(*tt_codes: str) -> None:
    """Metadata annotation — records which TestTarget (TT) codes are exercised.

    No effect on test pass/fail. Prints to stdout for capture in pytest -s.

    Per _specs/CROSS-FRAMEWORK-PARITY-EXECUTION-v0.1-CS.md §7.1 acceptance #2.

    Example:
        covers("TT-LOV-vehicleBrands", "TT-SCRN-rozcestnik")
    """
    print(f"[covers] {', '.join(tt_codes)}")
